pull.go

```diff
@@ -112,8 +112,9 @@ func (c *Client) fetch(ctx context.Context, rCtx *RemoteContext, ref string, lim
 		childrenHandler := images.ChildrenHandler(store)
 		// Set any children labels for that content
 		childrenHandler = images.SetChildrenLabels(store, childrenHandler)
-		// Filter children by platforms
-		childrenHandler = images.FilterPlatforms(childrenHandler, rCtx.PlatformMatcher)
+		// Filter manifests by platforms but allow to handle manifest
+		// and configuration for not-target platforms
+		childrenHandler = remotes.FilterManifestByPlatformHandler(childrenHandler, rCtx.PlatformMatcher)
```

Comment:
It looks like we are done this on purpose: 
1. we will download manifest for *all* platforms
2. but only will download the layers for the target 

However, it looks like it is little relationship with the target this PR was trying to do, may I know why we have this change?

**Example**: we will download all these files for busybox:

```
docker.io/library/busybox:latest:                                                 resolved      
index-sha256:061ca9704a714ee3e8b80523ec720c64f6209ad3f97c0ff7cb9ec7d19f15149f:    done          
manifest-sha256:aedc50f3ec56ef06f5cc2d6fc9959ea96249c6f0a8a010a2df916dce5ac0edc2: done          
manifest-sha256:4415a904b1aca178c2450fd54928ab362825e863c0ad5452fd020e92f7a6a47e: done          
manifest-sha256:b37971a5af7cc52577de16b0916f2f65ff84ed3852d5d0f3e6a8ab89b443eedb: done          
manifest-sha256:4275862e7069e6c520d7b1a906bc352a899d3ea93aa6da5ea1e0bca5bfa59a75: done          
manifest-sha256:5735bf09e9d95cf7704d4b1b10b52c7bdfc12312ba0855967f2a1ce998aa9c49: done          
manifest-sha256:9186030c9c6e59e0361838f610084b7d4f43f14854101a93dfe22d48f0e0dbf0: done          
manifest-sha256:a5e18b939c590ce6e91c6fd84063852ae1f83bc40689b544dc0cdafe85ff24ec: done          
manifest-sha256:59db9e5cc365f2f493317d3cf4847f02671fd88e9164acfbf464467df8f4cfdc: done          
config-sha256:de7af5d235954344bf4d47141efdd0268b8398c6b7f67948b1bae1c1e6f70446:   done          
config-sha256:2dfb9f9812cb6a4049ba0a706ac532438a78977c0d03cc49ca16040ed12ebf98:   done          
layer-sha256:697743189b6d255069caf6c455be10c7f8cae8076c6f94d224ae15cd41420e87:    done          
config-sha256:d8233ab899d419c58cf3634c0df54ff5d8acc28f8173f09c21df4a07229e1205:   done          
config-sha256:aafab271bd151b8d26f2aa9e59a2c3bd93d6a69d9cf73b93f93c94977647ef8a:   done          
config-sha256:076f0534356f9c34071c3160d3455611f655bca9b91a18b76bb1079f24045838:   done          
config-sha256:a1ba6c1cddb07a0bb2723e542ddc86e25606169504a5c194334a9e75f3a17716:   done          
config-sha256:3773eb31616a454b1b61ce270203f81a505242da04b371ead12efcce695cd2bc:   done          
config-sha256:79de0d36e9b31fd9f9d41bcaa07f281fb640a677ba31259b6f6facb0cbec1428:   done          
```